OpenRA.Mods.D2k/Traits/Buildings/DamagedWithoutFoundation.cs

```diff
@@ -52,7 +52,8 @@ public void AddedToWorld(Actor self)
 					safeTiles++;
 			}
 
-			damageThreshold = (info.DamageThreshold * health.MaxHP + (100 - info.DamageThreshold) * safeTiles * health.MaxHP / totalTiles) / 100;
+			if (totalTiles > 0)
```

Comment:
This will now cause the actor to lose 100 health on spawn if they have no cells occupied. You should add an else statement that throws an InvalidArgumentException.